workflow/controller/operator.go

```diff
@@ -1160,16 +1163,32 @@ func (woc *wfOperationCtx) checkParallelism(tmpl *wfv1.Template, node *wfv1.Node
 }
 
 func (woc *wfOperationCtx) executeContainer(nodeName string, tmpl *wfv1.Template, boundaryID string) *wfv1.NodeStatus {
+	var skipNodeInitialization bool
 	node := woc.getNodeByName(nodeName)
 	if node != nil {
-		return node
+		if strings.Contains(node.Message, exceededQuota) {
```

Comment:
Move the strings.Contains call into a function called ExceededQuota or something?